File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int left=0;
        int maxLen=0;
        int zero=0;
        for(int right=0;right<nums.size();right++){
            if(nums[right]==0){
                zero++;
            }

            while(zero > k){
                if(nums[left]==0){
                    zero--;
                }
                left++;
            }
            maxLen=max(maxLen,right-left+1);
        }
        return maxLen;
    }
};

File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int n=nums.length;
        int right=0;
        int maxLength=0;
        int zero=0;
        for(int left=0;left<n;left++){
            if(nums[left]==0){
                zero++;
            }
            while(zero >k){
                if(nums[right]==0){
                    zero--;
                }
                right++;
            }

            maxLength=Math.max(maxLength,left-right+1);
        }

        return maxLength;
    }
}